packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2389,19 +2391,32 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         "readonly",
       ];
       this.tsParseModifiers(
-        member,
+        modified,
         invalidModifersForStaticBlocks.concat(["static"]),
       );
 
+      for (const modifier of Object.keys(modified)) {
+        if (modifier === "accessibility") {
+          member.accessibility = modified[modifier];
+        } else if (modifier !== "static") {
+          member[modifier] = true;
+        }
+      }
```

Comment:
Object clone is also slow. Can we check `tt.braceL` when `modifier === "static"` in https://github.com/babel/babel/blob/fc66d4dd0572fe87a6c7169f9c3619e09a6acd33/packages/babel-parser/src/plugins/typescript/index.js#L306 ?

For cases like `static declare { }`, it doesn't matter if `static` is set because static block can not have any modifiers.

Reply:
We can check it in `parseModifiers`. But if we do that, we cannot know from the `parseClassMember` whether a member is static or not in

https://github.com/babel/babel/blob/fc66d4dd0572fe87a6c7169f9c3619e09a6acd33/packages/babel-parser/src/plugins/typescript/index.js#L2397-L2398

Reply:
It breaks some invalid test cases( static block has modifier ), but able to avoid deleting properties and copying objects by lookahead ( `lookaheadCharCode` ) them. What do you think? (f09c7f6, 57824b9)

Reply:
In that case we can have `parseModifiers` stopped at `static {` so we can still recover from cases when modifiers are on static block.

See also https://github.com/microsoft/TypeScript/blob/7a19c22063660da0b2f03fca97178679814de1e7/src/compiler/parser.ts#L6916-L6919

Reply:
Thanks for telling it me. I've fixed by referring to the TypeScript parser code ( d743bc8 )